Jenkinsfile

```diff
@@ -107,23 +107,18 @@ spec:
             }
         }
         stage('Deploy') {
-            when {
-              branch 'dev'
-            }
```

Comment:
```
when {
  branch 'dev'
}
```

Restoring these lines allows original behavior: gh-pages will only build the dev branch as it currently does.

If we can not use this feature build out our PR branches, I'd like to discuss if we could use some variation of this pattern.

Ideas:
- gh-pages also builds for test branch (Feature Branch - when we are working on a bigger effort)
- when there is a major update, we can use the pattern for still having example site of that version (most sites have this, including Bootstrap) 
<img width="265" alt="image" src="https://github.com/ASU/asu-unity-stack/assets/5209283/77c36188-464b-4eca-b241-53a43761e230">

Example: 
- https://unity.web.asu.edu/ 
- `https://unity.web.asu.edu/version/1.1.0/ `
  - this may not make since UDS packages have unique versioning
- `https://unity.web.asu.edu/release/2023q4/`
- etc.... (find a pattern that makes sense)